sql/core/src/test/scala/org/apache/spark/sql/Spark18601Suite.scala

```diff
@@ -0,0 +1,331 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.sql.catalyst.expressions.{Alias, Attribute, Cast, CreateArray, CreateMap, CreateNamedStruct, CreateNamedStructLike, CreateNamedStructUnsafe, Expression, GetArrayItem, GetArrayStructFields, GetMapValue, GetStructField, IntegerLiteral, Literal}
+import org.apache.spark.sql.catalyst.optimizer.Optimizer
+import org.apache.spark.sql.catalyst.plans.PlanTest
+import org.apache.spark.sql.catalyst.plans.logical.{LogicalPlan, Project}
+import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.test.SharedSQLContext
+import org.apache.spark.sql.types._
+import org.scalatest.{FunSuite, ShouldMatchers, Suite, Tag}
+
+import scala.collection.immutable.IndexedSeq
+import org.apache.spark.sql.catalyst.dsl.expressions._
+import org.apache.spark.sql.catalyst.dsl.plans._
+
+/**
+* Created by eyalf on 11/4/2016.
+*/
+class Spark18601Suite extends PlanTest with SharedSQLContext with ShouldMatchers{
+  //import testImplicits._
+
+  lazy val baseRelation = sqlContext.range( 1L, 1000L)
+  lazy val baseOptimizedPlan = baseRelation.queryExecution.optimizedPlan
+
+  val idRef = ('id).long.notNull
+  val idRefColumn = Column( "id" )
+  val struct1RefColumn = Column( "struct1" )
+
+  test( "explicit" ) {
+    val rel = baseRelation.select(
+      functions.struct( idRefColumn as "att" ).getField("att") as "outerAtt"
+    )
+    rel.schema shouldEqual
+      StructType( StructField( "outerAtt", LongType, nullable = false ) :: Nil )
+
+    val optimized = rel.queryExecution.optimizedPlan
+
+    val expected = baseOptimizedPlan.select(
+      idRef as "outerAtt"
+    )
+
+    comparePlans(optimized, expected)
+  }
+
+  test( "explicit - deduced att name") {
+    val rel = baseRelation.select(
+      functions.struct( idRefColumn as "att" ).getField("att")
+    )
+    rel.schema shouldEqual
+      StructType(
+        StructField( "named_struct(att, id AS `att`).att", LongType, nullable = false ) :: Nil
+      )
+
+    val optimized = rel.queryExecution.optimizedPlan
+
+    val expected = baseOptimizedPlan.select(
+      idRef as "named_struct(att, id AS `att`).att"
+    )
+
+    comparePlans(optimized, expected)
+  }
+
+  test( "collapsed" ) {

```

Comment:
Nit: `test( "collapsed" )` -> `test("collapsed")`

Please correct all the style issues in all the above test cases.